opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -176,3 +178,49 @@ def _shutdown():
         self._daemon_thread.join()
         self._exporter.shutdown()
         return True
+
+
+class PullMetricExporter(MetricExporter):
+    def __init__(self) -> None:
+        self._collect: Callable[
+            ["PullMetricExporter"], None,
+        ] = None
+
+    @final
+    def _set_collect_callback(
+        self,
+        func: Callable[
+            ["PullMetricExporter"], None,
+        ],
+    ) -> None:
+        """This function is internal to the SDK. It should not be called or overriden by users"""
+        self._collect = func
+
+    @final
+    def collect(self) -> None:
+        if self._collect:
+            self._collect()
+
+
+class PullMetricExporterReader(MetricReader):
```

Comment:
```suggestion
class PullMetricExporterMetricReader(MetricReader):
```

Reply:
Maybe this should be `PullExportingMetricReader`? I don't want to repeat Metric in the name. Thoughts?